### PR TITLE
Move threads e2e tests over to cypress

### PR DESCRIPTION
**Pull request by t3chguy**
Requires https://github.com/matrix-org/matrix-js-sdk/pull/2352/

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Review by kerryarchibald — APPROVED** [review]
LGTM! (and yay for removing the flaky puppeteer threads test, thanks!)

File: cypress/integration/2-login/consent.spec.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    it("should prompt the user to consent to terms when server deems it necessary", () => {
+        // Attempt to create a room using the js-sdk which should return an error with `M_CONSENT_NOT_GIVEN`
+        cy.window().then(win => {
+            win.mxMatrixClientPeg.matrixClient.createRoom({}).catch(() => {});
+
+            cy.stub(win, "open").as("windowOpen").returns({});
+        });
+
+        cy.get(".mx_QuestionDialog").within(() => {
```
Comment (t3chguy) [inline cypress/integration/2-login/consent.spec.ts:46]:
```suggestion
        // Accept terms & conditions
        cy.get(".mx_QuestionDialog").within(() => {
```

File: cypress/integration/2-login/consent.spec.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            cy.get("#mx_BaseDialog_title").contains("Terms and Conditions");
+            cy.get(".mx_Dialog_primary").click();
+        });
+
+        cy.get<SinonStub>("@windowOpen").then(stub => {
+            const url = stub.getCall(0).args[0];
+
+            cy.origin(synapse.baseUrl, { args: { url } }, ({ url }) => {
```
Comment (t3chguy) [inline cypress/integration/2-login/consent.spec.ts:54]:
```suggestion
            // Go to Synapse's consent page and accept it
            cy.origin(synapse.baseUrl, { args: { url } }, ({ url }) => {
```

File: cypress/global.d.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+declare global {
+    // eslint-disable-next-line @typescript-eslint/no-namespace
+    namespace Cypress {
+        interface ApplicationWindow {
+            mxMatrixClientPeg: {
+                matrixClient?: MatrixClient;
+            };
+            mxDispatcher: MatrixDispatcher;
+            beforeReload?: boolean; // for detecting reloads
+            // TODO
```
Comment (t3chguy) [inline cypress/global.d.ts:31]:
```suggestion
            // Partial type for the matrix-js-sdk module, exported by browser-matrix
```

**Pull request merged**